src/borg/patterns.py

```diff
@@ -150,9 +150,9 @@ def match(self, path):
 
         if value is not non_existent:
             # we have a full path match!
-            # TODO: get from pattern; don't hard-code
-            self.recurse_dir = True
-            return value
+            self.recurse_dir = (value is not IECommand.ExcludeNoRecurse)
+
```

Comment:
hmm, an empty line here is not really needed/helpful, is it?

also, can we do the same check at both places in the same way? (see my previous comment)